Assets/GothicVR/Scripts/Npc/Actions/AnimationActions/UseItemToState.cs

```diff
@@ -1,55 +1,88 @@
 using GVR.Caches;
 using GVR.Creator;
+using GVR.GothicVR.Scripts.Manager;
 using UnityEngine;
+using ZenKit.Daedalus;
 
 namespace GVR.Npc.Actions.AnimationActions
 {
     public class UseItemToState : AbstractAnimationAction
     {
         
-        private const string animationStartScheme = "T_{0}_STAND_2_S{1}" ;
-        private const string animationEndScheme = "T_{0}_S{1}_2_STAND" ;
-        
+        private const string animationScheme = "T_{0}_{1}_2_{2}";
+        private const string LoopAnimationScheme = "S_{0}_S{1}";
+
+        private int itemToUse => Action.Int0;
+        private int desiredState => Action.Int1;
+
+
         public UseItemToState(AnimationAction action, GameObject npcGo) : base(action, npcGo)
         { }
 
         public override void Start()
         {
-            // Nothing in hand && new item shall be put into hand
-            if (Props.itemAnimationState < 0 && Action.Int1 >= 0)
+            PhysicsHelper.DisablePhysicsForNpc(Props);
+            PlayTransitionAnimation();
+        }
+
+        private void PlayTransitionAnimation()
+        {
+            int oldItemAnimationState = Props.itemAnimationState;
+            int newItemAnimationState;
+            if (desiredState > Props.itemAnimationState)
             {
-                StartItemStateAnimation(Action.Int1);
                 Props.hasItemEquipped = true;
+                Props.currentItem = itemToUse;
+                newItemAnimationState = ++Props.itemAnimationState;
             }
-            // Something in hand && item shall be removed
-            else if (Props.itemAnimationState >= 0 && Action.Int1 < 0)
+            else
             {
-                EndItemStateAnimation(Props.itemAnimationState);
                 Props.hasItemEquipped = false;
+                Props.currentItem = -1;
+                // e.g. Babe brush doesn't call it automatically. We therefore need to force remove the brush item from hand.
```

Comment:
That's pretty specific. How did you come up with that finding? Is that a bug / problem in the original game? Is it a proper solution?